numpy/core/src/multiarray/compiled_base.c

```diff
@@ -315,7 +315,8 @@ arr_insert_loop(char *mptr, char *vptr, char *input_data, char *zero,
                 int totmask, int numvals, int nd, npy_intp *instrides,
                 npy_intp *inshape)
 {
-    int mindx, rem_indx, indx, i, copied;
+    int mindx, rem_indx, i, copied;
```

Comment:
@jaimefrio  Good point - actually I think that `totmask` and `numvals` should also be `npy_intp` (otherwise I still see failures for ubyte arrays larger than 2**31)